### PR TITLE
support reward modeling and ppo 

### PR DESCRIPTION
**Pull request by hjh0119**
# PR type
- [x] New Feature

# PR information

support `reward modeling` for LLM and MLLM
support `PPO` for LLM

### CODE REVIEW

**Review by Jintao-Huang — APPROVED** [review]

File: swift/trainers/mixin.py [hunk 1]
```diff
@@ -98,7 +99,7 @@ def __init__(self,
             optimizers=optimizers,
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
             **kwargs)
-        if not self.label_names:
+        if not hasattr(self, 'label_names') or not self.label_names:
```
Comment (hjh0119) [inline swift/trainers/mixin.py:102]: PPOv2Trainer does not have the `label_names` attribute.

File: swift/llm/tuner.py [hunk 2]
```diff
@@ -98,7 +98,8 @@ def prepare_model(model, args: SftArguments):
         if args.resume_from_checkpoint is None:
             handle_target_modules(model, args)
             handle_modules_to_save(model, args)
-            if args.init_lora_weights and args.init_lora_weights.lower() in ('true', 'false'):
+            if args.init_lora_weights and isinstance(args.init_lora_weights,
+                                                     str) and args.init_lora_weights.lower() in ('true', 'false'):
```
Comment (hjh0119) [inline swift/llm/tuner.py:102]: Avoid errors in reward_model_args.

File: swift/llm/utils/argument.py [hunk 3]
```diff
@@ -1196,6 +1196,10 @@ def _init_training_args(self) -> None:
         if 'accelerator_config' in parameters:
             kwargs['accelerator_config'] = {'dispatch_batches': False}
 
+        metric_for_best_model = 'rouge-l' if self.predict_with_generate else 'loss'
+        if hasattr(self, 'rlhf_type') and self.rlhf_type == 'ppo':
+            metric_for_best_model = None
```
Comment (hjh0119) [inline swift/llm/utils/argument.py:1201]: In the PPO training metrics, there are no metrics that start with "eval", set None here

**Review by Jintao-Huang — APPROVED** [review]

**Review by tastelikefeet — APPROVED** [review]
